### PR TITLE
Selector one-frame delay in 4.24

### PR DESCRIPTION
**Pull request by josator**
I created this branch for the selector one frame delay on 4.24, I made the code , but I don't have that unreal engine version installed, please check it and test it.

### CODE REVIEW

File: Source/UINavigation/Private/UINavWidget.cpp [hunk 1]
```diff
@@ -594,6 +591,8 @@ void UUINavWidget::UINavSetup()
 	{
 		OnPromptDecided(PromptWidgetClass, PromptSelectedIndex);
 	}
+
+	bCompletedSetup = true;
```
Comment (goncasmage1) [inline Source/UINavigation/Private/UINavWidget.cpp:595]: This should be true before OnSetupCompleted is called

File: Source/UINavigation/Private/UINavWidget.cpp [hunk 2]
```diff
@@ -1263,20 +1279,19 @@ void UUINavWidget::ReplaceButtonInNavigationGrid(UUINavButton * ButtonToReplace,
...
+		UpdateSelectorPrevButtonIndex = ButtonIndex;
+		UpdateSelectorNextButtonIndex = NewCurrentButton->ButtonIndex;
+		bUpdateSelector = true;
 	}
 
+	ButtonIndex = NewCurrentButton->ButtonIndex;
+	
 	for (UScrollBox* ScrollBox : ScrollBoxes)
 	{
-		if (NewCurrentButton->IsChildOf(ScrollBox))
```
Comment (goncasmage1) [inline Source/UINavigation/Private/UINavWidget.cpp:1275]: Why did you remove this?

File: Source/UINavigation/Public/UINavWidget.h [hunk 3]
```diff
@@ -31,19 +31,24 @@ class UINAVIGATION_API UUINavWidget : public UUserWidget
 	bool bCompletedSetup = false;
 	bool bSetupStarted = false;
 
-	bool bShouldTick = true;
 	bool bMovingSelector = false;
 	bool bIgnoreMouseEvent = false;
 	bool bReturning = false;
+		
+	bool bUINavSetup = false;
```
Comment (goncasmage1) [inline Source/UINavigation/Public/UINavWidget.h:38]: Rename this to something like bShouldTickUINavSetup

**Review by goncasmage1 — CHANGES_REQUESTED** [review]

**Comment by goncasmage1** [comment]
Also there are too many reviews 😅 I don't know which ones I should be taking a look at!

File: Source/UINavigation/Private/UINavWidget.cpp (hunk 1 above)
Comment (josator) [reply Source/UINavigation/Private/UINavWidget.cpp:595]: ok

File: Source/UINavigation/Public/UINavWidget.h (hunk 3 above)
Comment (josator) [reply Source/UINavigation/Public/UINavWidget.h:38]: ok

File: Source/UINavigation/Private/UINavWidget.cpp (hunk 2 above)
Comment (josator) [reply Source/UINavigation/Private/UINavWidget.cpp:1275]: Reading Unreal engine source code, SSCrollBox.cpp function bool SScrollBox::InternalScrollDescendantIntoView(), called from ScrollDescendantIntoView(), called from ScrollBox.cpp ScrollWidgetIntoView() is already checking if the widget is a descendant of the scrollbox and printing a warning message if it isn't. So The check is performed twice for each button descendant of the scrollbox.

Comment (josator) [reply Source/UINavigation/Private/UINavWidget.cpp:1275]: I left it as it was to avoid the annoying message.

**Review by goncasmage1 — APPROVED** [review]

**Comment by goncasmage1** [comment]
Done 😄 thanks a lot for the PR! 💪 Good job.

**Comment by josator** [comment]
Did not read  :-)
